### PR TITLE
feat(session-replay): adds banner for session replay waitlist

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a banner to the logic page to show a banner with a link to join the waitlist for Session Replay. Note that we purposely don't show this to self-hosted users.

Screen Shot 2022-10-31 at 10 36 05 PM

I added a CSS style to hide the banner if the screen is so narrow that it wouldn't fit on one line.

### CODE REVIEW

File: src/sentry/templates/sentry/layout.html [hunk 1]
```diff
@@ -71,6 +71,33 @@
...
+      line-height: 1.5;
+      display: inline-flex;
+      align-items: center;
+      justify-content: center;
+      text-align: left;
+      flex-grow: 1;
+      font-size: 16px;
+      font-family: Rubik;
+    }
+  </style>
```
Comment (billyvg) [inline src/sentry/templates/sentry/layout.html:98]: Do we not have any existing CSS for this?

File: src/sentry/web/frontend/auth_login.py [hunk 2]
```diff
@@ -262,7 +263,9 @@ def handle_basic_auth(self, request: Request, **kwargs):
             "register_form": register_form,
             "CAN_REGISTER": can_register,
             "join_request_link": self.get_join_request_link(organization),
+            "show_session_replay_banner": not is_self_hosted(),  # show to everyone except self-hosted
```
Comment (billyvg) [inline src/sentry/web/frontend/auth_login.py:266]: I'm assuming ST is included in self-hosted?

**Review by billyvg — APPROVED** [review]
This yellow tho...

File: src/sentry/web/frontend/auth_login.py (hunk 2 above)
Comment (scefali) [reply src/sentry/web/frontend/auth_login.py:266]: @billyvg good question. I will double check.

File: src/sentry/templates/sentry/layout.html (hunk 1 above)
Comment (scefali) [reply src/sentry/templates/sentry/layout.html:98]: @billyvg Not in HTML that's not used in emails. Not too many styles to choose from there.

**Pull request merged**